public/omni/cluster-management/scale-your-cluster/cluster-autoscaler.mdx

```diff
@@ -0,0 +1,361 @@
+## Step 2: Create control plane machines from your images
+
+Use the control plane image to create your control plane machines.
+
+For high availability, we recommend creating at least three control plane machines.
+
+<Note>
+We do not recommend horizontally autoscaling control plane machines.
+If your control plane needs more capacity, scale **vertically** by using larger machine types rather than adding more control plane nodes.
+
+Adding additional nodes increases the number of etcd replicas, which can slow down write performance.
+</Note>
+
+When you're done, all of your machines should appear in the **Machines** section of the Omni dashboard.
+
+## Step 3: Create Machine Classes
+
+[Machine Classes](../../omni-cluster-setup/create-a-machine-class) group machines by labels and act as the pool description for clusters.
+
+You’ll create two Machine Classes:
+
+- one for control plane machines
+- one for worker machines
+
+### 3.1: Control plane Machine Class
+
+To create a machine class for your control plane machines:
+
+1. Create a file named `controlplane-machine-class.yaml`:
+
+```yaml
+metadata:
+  namespace: default
+  type: MachineClasses.omni.sidero.dev
+  id: cluster-autoscaler-controlplane
+spec:
+  matchlabels:
+    # Match the label you applied to your control plane AMI
+    - role = autoscaler-controlplane-machine
+```
+
+This creates a `cluster-autoscaler-controlplane` machine class with your control plane AMI label ( e.g., `role:autoscaler-controlplane-machine`).
+
+2. Apply it:
+
+```bash
+omnictl apply -f controlplane-machine-class.yaml
+```
+
+3. Verify it exists:
+
+```bash
+omnictl get machineclasses
+```
+
+### 3.2: Worker Machine Class
+
+Repeat the process for the worker machines:
+
+1. Create a file named `worker-machine-class.yaml`:
+
+```bash
+metadata:
+  namespace: default
+  type: MachineClasses.omni.sidero.dev
+  id: cluster-autoscaler-worker
+spec:
+  matchlabels:
+    # Match the label you applied to your worker AMI
+    - role = autoscaler-worker-machine
+```
+
+2. Apply it:
+
+```bash
+omnictl apply -f worker-machine-class.yaml
+```
+
+3. Verify:
+
+```bash
+omnictl get machineclasses
+```
+
+## Step 4: Create a cluster using a template
+
+Next, create a cluster that uses the Machine Classes you defined in Step 3.
+
+1. Create a file named `cluster-template.yaml` with the following content, updating each placeholder:
+
+```yaml
+# cluster-template.yaml
+kind: Cluster
+name: <cluster-name> # Replace with your cluster name
+kubernetes:
+  version: v1.34.1    # Replace with your desired Kubernetes version
+talos:
+  version: v1.11.3    # Replace with your desired Talos version
+
+---
+kind: ControlPlane
+machineClass:
+  name: <controlplane-class-name> # e.g. cluster-autoscaler-controlplane
+  size: <controlplane-count>      # e.g. 3
+
+---
+kind: Workers
+machineClass:
+  name: <worker-class-name>       # e.g. cluster-autoscaler-worker
+  size: unlimited
+```
+
+Replace:
+
+- `<cluster-name>` — your cluster name
```

Comment:
Should we make these environment variables at the top of the guide and then make the other commands copy pastable?